pkg/reconciler/v1alpha1/revision/resources/deploy_test.go

```diff
@@ -39,12 +39,219 @@ import (
+		}, {
+			Name:  "SERVING_REVISION",
+			Value: "bar", // matches name
+		}, {
+			Name:  "SERVING_AUTOSCALER",
+			Value: "autoscaler", // no autoscaler configured.
+		}, {
+			Name:  "SERVING_AUTOSCALER_PORT",
+			Value: "8080",
+		}, {
+			Name:  "CONTAINER_CONCURRENCY",
+			Value: "0",
+		}, {
+			Name:  "REVISION_TIMEOUT_SECONDS",
+			Value: "45",
+		}, {
+			Name: "SERVING_POD",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{FieldPath: "metadata.name"},
+			},
+		}, {
+			Name: "SERVING_LOGGING_CONFIG",
+			// No logging configuration
+		}, {
+			Name: "SERVING_LOGGING_LEVEL",
+			// No logging level
+		}, {
+			Name:  "USER_PORT",
+			Value: "8080",
+		}, {
+			Name:  "SYSTEM_NAMESPACE",
+			Value: system.Namespace(),
+		}},
+	}
+
+	defaultFluentdContainer = &corev1.Container{
+		Name:      FluentdContainerName,
+		Image:     "indiana:jones",
+		Resources: fluentdResources,
+		Env: []corev1.EnvVar{{
+			Name:  "FLUENTD_ARGS",
+			Value: "--no-supervisor -q",
+		}, {
+			Name:  "SERVING_CONTAINER_NAME",
+			Value: UserContainerName,
+		}, {
+			Name: "SERVING_CONFIGURATION",
+			// No owner reference
+		}, {
+			Name:  "SERVING_REVISION",
+			Value: "bar",
+		}, {
+			Name:  "SERVING_NAMESPACE",
+			Value: "foo",
+		}, {
+			Name: "SERVING_POD_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.name",
+				},
+			},
+		}},
+		VolumeMounts: fluentdVolumeMounts,
+	}
+
+	defaultPodSpec = &corev1.PodSpec{
+		Volumes:                       []corev1.Volume{varLogVolume},
+		TerminationGracePeriodSeconds: refInt64(45),
+	}
+
+	defaultDeployment = &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "foo",
+			Name:      "bar-deployment",
+			Labels: map[string]string{
+				serving.RevisionLabelKey: "bar",
+				serving.RevisionUID:      "1234",
+				AppLabelKey:              "bar",
+			},
+			Annotations: map[string]string{},
+			OwnerReferences: []metav1.OwnerReference{{
+				APIVersion:         v1alpha1.SchemeGroupVersion.String(),
+				Kind:               "Revision",
+				Name:               "bar",
+				UID:                "1234",
+				Controller:         &boolTrue,
+				BlockOwnerDeletion: &boolTrue,
+			}},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &one,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					serving.RevisionUID: "1234",
+				},
+			},
+			ProgressDeadlineSeconds: &ProgressDeadlineSeconds,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						serving.RevisionLabelKey: "bar",
+						serving.RevisionUID:      "1234",
+						AppLabelKey:              "bar",
+					},
+					Annotations: map[string]string{
+						sidecarIstioInjectAnnotation: "true",
+					},
+				},
+				// Spec: filled in by makePodSpec
+			},
+		},
+	}
 )
 
 func refInt64(num int64) *int64 {
 	return &num
 }
 
+type containerModifier func(*corev1.Container)
```

Comment:
`containerOpt(ion)`, `podSpecOpt`, `deploymentOpt` -- is the naming I think we use else were (e.g. functional.go).